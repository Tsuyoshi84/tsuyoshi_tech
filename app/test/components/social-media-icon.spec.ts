import { mount } from '@vue/test-utils'
import SocialMediaIcon from '@/components/social-media-icon.vue'

const getComponent = () => {
  return mount(SocialMediaIcon, {
    propsData: { href: 'https://example.com', color: 'red' },
    slots: {
      default: '<div>slot-content</div>'
    },
    stubs: {}
  })
}

describe('SocialMediaIcon', () => {
  test('is a Vue instance', () => {
    const wrapper = getComponent()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  describe('when not hover', () => {
    test('should use default color', done => {
      const wrapper = getComponent()
      expect(wrapper.html()).toMatchSnapshot()
      done()
    })
  })

  describe('when hover', () => {
    test('should use given color', done => {
      const wrapper = getComponent()
      wrapper.trigger('mouseover')
      wrapper.vm.$nextTick(() => {
        expect(wrapper.html()).toMatchSnapshot()
        done()
      })
    })
  })
})
