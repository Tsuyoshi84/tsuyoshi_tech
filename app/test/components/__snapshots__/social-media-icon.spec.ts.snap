// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaIcon when hover should use given color 1`] = `
<div class="box"><a href="https://example.com" alt="Example" class="link-icon" style="color: red;">
    <div>slot-content</div>
  </a></div>
`;

exports[`SocialMediaIcon when not hover should use default color 1`] = `
<div class="box"><a href="https://example.com" alt="Example" class="link-icon" style="color: currentColor;">
    <div>slot-content</div>
  </a></div>
`;
